README.md

```diff
@@ -289,19 +289,28 @@ By default, both `pip-compile` and `pip-sync` will look first
 for a `.pip-tools.toml` file and then in your `pyproject.toml`. You can
 also specify an alternate TOML configuration file with the `--config` option.
 
-For example, to by default generate `pip` hashes in the resulting
-requirements file output, you can specify in a configuration file
+For example, to by default annotate dependencies with their origins in the resulting
+requirements file output, you can specify in a configuration file:
 
 ```toml
 [tool.pip-tools]
-generate-hashes = true
+annotate = true
 
 ```
 
 Options to `pip-compile` and `pip-sync` that may be used more than once
 must be defined as lists in a configuration file, even if they only have one
 value.
 
+Since `pip-tools` supports default values vor [all valid command-line flags](/cli/index.md)
+of its subcommands, you can also achieve the functionality in the above example
+by setting a configuration value for the negative flag `--no-annotate`:
+
+```toml
+[tool.pip-tools]
+no-annotate = false
+```
+
```

Comment:
This may seem like a special case. However, we essentially support any long options (those starting with `--` prefix) listed in [command line reference](https://pip-tools.readthedocs.io/en/latest/cli/) represented in either snake or kebab case (without the double dash). Perhaps it's worth mentioning this instead? What do you think?

Reply:
Apparently, the Sphinx TOML lexer doesn't seem to concur with that suggestion...

Reply:
See https://github.com/jazzband/pip-tools/pull/1913#discussion_r1272724649

Reply:
I've tested that configuration though, and seen it work...

Reply:
It might work, since we do the normalization after the validation for now. However, we should not encourage this, because it will be fixed soon.

Reply:
We may be able to call `_validate_config()` directly in `_invert_negative_bool_options_in_config()`.